prometheus/src/main/scala/filodb/prometheus/ast/Base.scala

```diff
@@ -1,26 +1,48 @@
 package filodb.prometheus.ast
 
-import filodb.query.{PeriodicSeriesPlan, RawSeriesPlan}
+import filodb.query._
+
+sealed trait TimeParams {
+  def start: Long   // in seconds sine Epoch
+  def end: Long
+  def step: Long
+}
 
 /**
  * NOTE: start and end are in SECONDS since Epoch
  */
-case class QueryParams(start: Long, step: Long, end: Long)
+final case class QueryParams(start: Long, step: Long, end: Long) extends TimeParams
```

Comment:
Can rename trait as TimeRangeParam
QueryParams -> TimeParam  (QueryParam is not specific at all)

Reply:
Renamed, `QueryParams` -> `TimeStepParams`